apps/server/src/modules/tool/external-tool/uc/external-tool.uc.ts

```diff
@@ -71,4 +96,29 @@ export class ExternalToolUc {
 		const user: User = await this.authorizationService.getUserWithPermissions(userId);
 		this.authorizationService.checkAllPermissions(user, [permission]);
 	}
+
+	async getExternalToolBinaryLogo(toolId: EntityId): Promise<ExternalToolLogo> {
+		const tool: ExternalTool = await this.externalToolService.findExternalToolById(toolId);
+
+		if (!tool.logoBase64) {
+			throw new ExternalToolLogoNotFoundLoggableException(toolId);
+		}
+
+		const logoBinaryData: Buffer = Buffer.from(tool.logoBase64, 'base64');
+
+		const externalToolLogoDto: ExternalToolLogo = new ExternalToolLogo({
+			contentType: this.detectContentType(logoBinaryData),
+			logo: logoBinaryData,
+		});
+
+		return externalToolLogoDto;
+	}
+
+	private detectContentType(imageBuffer: Buffer): string {
+		const imageSignature: string = imageBuffer.toString('hex', 0, 3);
+
+		const contentType: string = contentTypeDetector[imageSignature] || 'image/png';
```

Comment:
i would not use a default like this here. If the Image is the wrong format, then nothing should be displayed and an error thrown.

Reply:
Interestingly, the browser can handle it. I could also always put image/png ... Lets talk about it

Reply:
i set it to octet stream as default for now